components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/utils/AuthUtil.java

```diff
@@ -162,23 +155,19 @@ public static String extractTokenFromHeaders(Request request, String cookieHeade
      * @return Cookie object.
      */
     public static NewCookie cookieBuilder(String name, String value, String path, boolean isSecure,
-                                          boolean isHttpOnly, String expiresIn) {
+                                          boolean isHttpOnly, String expiresIn, String environmentName) {
         StringBuilder stringBuilder = new StringBuilder();
-        stringBuilder.append(value).append(COOKIE_PATH_SEPERATOR).append(path).append(COOKIE_VALUE_SEPERATOR);
+        stringBuilder.append(value).append(COOKIE_PATH_SEPARATOR).append(path).append(COOKIE_VALUE_SEPARATOR);
+        if (expiresIn != null && !expiresIn.isEmpty()) {
```

Comment:
We can use  StringUtils.isEmpty(expiresIn) to check for both null and empty. You can replace the condition with ! StringUtils.isEmpty(expiresIn)